### PR TITLE
T4793: Added warning about disable-route-autoinstall

### PR DESCRIPTION
**Pull request by aapostoliuk**
## Change Summary

Added warning message about disable-route-autoinstall when ipsec vti is used.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4793

## Component(s) name

ipsec

## Proposed changes

Added warning message about disable-route-autoinstall when ipsec vti is used.

Create vpn ipsec tunnel using vti interface but do not use the next command
`set vpn ipsec options disable-route-autoinstall`
then
```
vyos@vyos# commit
[ vpn ipsec ]
WARNING: When using site-to-site IPsec with VTI interfaces,
it is recommended to use

 <set vpn ipsec options disable-route-autoinstall>

loaded ike secret 'ike_TESTPEER'
loaded connection 'TESTPEER'
successfully loaded 1 connections, 0 unloaded

[edit]

```
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/conf_mode/vpn_ipsec.py [hunk 1]
```diff
@@ -438,6 +438,12 @@ def verify(ipsec):
                 if 'local_address' in peer_conf and 'dhcp_interface' in peer_conf:
                     raise ConfigError(f"A single local-address or dhcp-interface is required when using VTI on site-to-site peer {peer}")
 
+                tmp = dict_search('options', ipsec)
+                if not tmp or 'disable_route_autoinstall' not in tmp:
+                    print(f'WARNING: When using site-to-site IPsec with VTI '
```
Comment (c-po) [inline src/conf_mode/vpn_ipsec.py]: Please use: `vyos.base.Warning()` instead as it has build in string formatting.

**Review by c-po — CHANGES_REQUESTED** [review]

File: src/conf_mode/vpn_ipsec.py (hunk 1 above)
Comment (aapostoliuk) [reply src/conf_mode/vpn_ipsec.py]: I tried to use it. But it is impossible if I want to use a warning message of more than one line using \n. Or I need to use many spaces.

Comment (c-po) [reply src/conf_mode/vpn_ipsec.py]: `vyos.base.Warning()` does automatic split and new-line handling after 72 characters. Please use it.

Comment (c-po) [reply src/conf_mode/vpn_ipsec.py]: Improved version of `vyos.base.Warning()` was merged from https://github.com/vyos/vyos-1x/pull/1660, can you update this PR so we can merge it?

Comment (aapostoliuk) [reply src/conf_mode/vpn_ipsec.py]: done

**Review by c-po — APPROVED** [review]

**Pull request merged**